ConfigSpace/hyperparameters.pyx

```diff
@@ -2556,7 +2556,14 @@ cdef class CategoricalHyperparameter(Hyperparameter):
             Probability density values of the input vector
         """
         probs = np.array(self.probabilities)
+        vector = np.atleast_1d(vector)
+        nan = np.isnan(vector)
+        if np.any(nan):
+            vector[nan] = 0
```

Comment:
As the res[nan] is set to zero further belows, this appears if-block appears to be unnecessary.

Reply:
The problem is that vector is used in the next line to access the probabilities from `probs`. We need any valid index for `probs`. Therefore, I just picked `0`. I added a comment to make this clear